File: core/service.go
package core

import (
	"github.com/louisevanderlith/husk/validation"
	"time"
)

type Service struct {
	Duration    time.Duration
	Location    string
	Description string `hsk:"size(256)"`
}

func (o Service) Valid() error {
	return validation.Struct(o)
}

File: lib/bodies/service.dart
class Service {
  final int hours;
  final int minutes;
  final int seconds;
  final String location;
  final String description;

  Service(this.location, this.description, this.hours, this.minutes, this.seconds);

  Map<String, dynamic> toJson() {
    return {
      "Duration": "${hours}h${minutes}m${seconds}s",
      "Location": location,
      "Description": description,
    };
  }
}
